### PR TITLE
added doc check to all .rs files

### PR DESCRIPTION
**Pull request by chutchi2**
#85 
Added in rust macro to check docs in all .rs files found in the project.

### CODE REVIEW

**Comment by alphaville** [comment]
@chutchi2 thank you for your contribution. You don't need to include `#![deny(missing_docs)]` in all files in the project. It suffices to do so in the main `lib.rs`. We hadn't done so so far because we were planning to remove the `continuation` module. I have done so in PR #133. Would you like to add `#![deny(missing_docs)]` and submit a pull request to branch `dev/rm-homotopy`?

By the way, the unit tests fail. You can run them locally with `cargo test`.

File: src/tests.rs [hunk 1]
```diff
@@ -1,3 +1,4 @@
+#![deny(missing_docs)]
```
Comment (alphaville) [inline src/tests.rs]: There shouldn't be a `#![deny(missing_docs)]` in test files.

File: src/alm/alm_cache.rs [hunk 2]
```diff
@@ -1,3 +1,4 @@
+#![deny(missing_docs)]
```
Comment (alphaville) [inline src/alm/alm_cache.rs]: no need to include this in all files; you just need this once in `src/lib.rs`

File: src/tests.rs (hunk 1 above)
Comment (chutchi2) [reply src/tests.rs]: woops, clearly i tried to do this way too fast

File: src/alm/alm_cache.rs (hunk 2 above)
Comment (chutchi2) [reply src/alm/alm_cache.rs]: okay, i'll be a wee bit more thorough this time

**Pull request closed without merging**